### PR TITLE
chore: use private fields

### PR DESCRIPTION
**Pull request by jrandolf-google**
This PR changes all underscored fields/methods into private fields and underscores (and labels as internal) all cross-module, camelCase, internal fields/variables/methods.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,3 +1,12 @@
+// TODO: Enable this at some point.
```
Comment (OrKoN) [inline .eslintrc.js:1]: what prevents it from being enabled right now?

**Review by OrKoN — APPROVED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (jrandolf) [reply .eslintrc.js:1]: There are some exceptions still.

**Comment by smashah** [comment]
I understand that this is js convention, however what other reasoning is there for this PR? Lots of dependent projects need to use `HTTPRequest._client` or `Page._client`.

**Comment by OrKoN** [comment]
@smashah to maintain the project we want to clearly separate the public API from internals. In this case, dependent projects were using the undocumented internal API. Please file a issue for your use case and we will try to expose the API in a documented and sustainable way. Btw, the `Page._client()` is still there.

**Comment by smashah** [comment]
Thanks for the explanation @OrKoN 

I have submitted https://github.com/puppeteer/puppeteer/pull/8545 to resolve an urgent use case

**Comment by casey6** [comment]
I was a little disappointed to discover that this breaking change made it into a minor release (`14.4.0`) completely undocumented and missing from the release notes.

I have some simple code that proxies a Page instance, which now throws a type error `Cannot read private member from an object whose class did not declare it`. 😞

**Comment by jrandolf-google** [comment]
@casey6 This is not a breaking change unless you use internal code which is **always** subject to breaking changes. If there is public behavior that we amended, please submit an issue and we can try to accommodate. That being said, we do have some places unlabeled with `@internal`, so that also might have been the problem. If such is the case, we apologize.

**Comment by casey6** [comment]
I'm simply proxying the page instance, not using any internal code:

```js
const proxiedPage = new Proxy(page, {
  get(page, property) {
    // intercept screenshot calls to override the path property
    if (property === 'screenshot') {
      return async (options = {}) => {
        const filename = options.path?.split('/').pop() ?? `${Date.now()}.png`

        return await page.screenshot({
          ...options,
          path: `/path/to/${filename}`
        })
      }
    }

    return page[property]
  }
})
```

**Comment by jrandolf-google** [comment]
Ah, I see... This is a well-known issue: https://github.com/tc39/proposal-class-fields/issues/106. If I recall, there is a solution using inheritance.

**Pull request merged**